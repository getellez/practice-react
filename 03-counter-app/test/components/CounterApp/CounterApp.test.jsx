import { fireEvent, getByText, render, screen } from "@testing-library/react";
import { CounterApp } from "../../../src/components/CounterApp/CounterApp";

describe('<CounterApp> component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<CounterApp value={100} />)
    expect(container).toMatchSnapshot()
  })
  it('should show 100 as an initial value', () => {
    render(<CounterApp value={100} />)
    expect(screen.queryByTestId('counter-value').innerHTML).toContain("100")
  });
  
  it('should add +1 ', () => {
    render(<CounterApp value={100} />)
    fireEvent.click(screen.getByText('+ 1'))
    expect(screen.getByText('101')).toBeTruthy()
  });

  it('should substract 1 ', () => {
    render(<CounterApp value={100} />)
    fireEvent.click(screen.getByText('- 1'))
    expect(screen.getByText(99)).toBeTruthy()
  });

  it('should reset the value', () => {
    render(<CounterApp value={100} />)
    fireEvent.click(screen.getByText('+ 1'))
    fireEvent.click(screen.getByText('+ 1'))
    fireEvent.click(screen.getByText('+ 1'))
    fireEvent.click(screen.getByText('Reset'))
    expect(screen.getByText(100)).toBeTruthy()
  });

})