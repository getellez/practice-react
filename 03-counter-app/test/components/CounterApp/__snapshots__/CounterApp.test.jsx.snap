// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterApp> component should match the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="counter-value"
  >
    100
  </h2>
  <button>
    + 1
  </button>
  <button>
    - 1
  </button>
  <button>
    Reset
  </button>
</div>
`;
